pennylane/qnode.py

```diff
@@ -616,6 +619,40 @@ def __call__(self, *args, **kwargs):
         )
         self._tape_cached = using_custom_cache and self.tape.hash in cache
 
+        if qml.query_return():
+            res = qml.execute_new(
+                [self.tape],
+                device=self.device,
+                gradient_fn=self.gradient_fn,
+                interface=self.interface,
+                gradient_kwargs=self.gradient_kwargs,
+                override_shots=override_shots,
+                **self.execute_kwargs,
+            )
+
+            if not isinstance(self._qfunc_output, tuple) and not isinstance(
+                self._qfunc_output, qml.measurements.MeasurementProcess
+            ):
+                if not self.device._shot_vector:
+                    res = type(self.tape._qfunc_output)(res[0])

```

Comment:
Does `self._qfunc_output` differ from `self.tape._qfunc_output`?

Reply:
No I think it is similar, I will make it coherent.